### DISCUSSION TITLE
Only write MSH empty segment. Missing PV1, OBR

### DISCUSSION THREAD

**Discussion by Migas1979** [nextgenhealthcare/connect#4562] [Q&A]
Hi,

In destination transform I have:

msg['MSH']['MSH.3']['MSH.3.1']='ICS';
msg['MSH']['MSH.4']['MSH.4.1']='HIS';
msg['MSH']['MSH.7']['MSH.7.1']=currentdate;

msg['PID']['PID.2']['PID.2.1']='1'
msg['PID']['PID.3']['PID.3.1']=pid;

msg['PV1']['PV1.2']['PV1.2.1']='I';
msg['PV1']['PV1.3']=bed;

msg['OBR']['OBR.4']='34566-0^SIGNOS VITALES';

If I do a logger.error(msg);
I have the predicted result.

But in my filewriter I am only getting the empty MSH segment.
MSH|^~\&|ICS||NUR||20210522200||ORU^R01|10072020530100004593|P|2.5||||AL|ER

Do you have any idea about what I am doing wrong?

**Comment by tonygermano** [comment]
Your MSH.4 is also blank instead of what you are setting it to. Do you have an outbound template in the transformer that is exactly that MSH segment? If so, delete it. Mirth is looking for you to make modifications to `tmp` instead of `msg` when it is present.
